File: app/src/main/java/ch/unstable/ost/api/StationsDAO.java
package ch.unstable.ost.api;


import android.support.annotation.Nullable;

import java.io.IOException;

public interface StationsDAO {
    ch.unstable.ost.api.model.impl.Location[] getStationsByQuery(String query) throws IOException;

    ch.unstable.ost.api.model.impl.Location[] getStationsByQuery(String query, @Nullable ch.unstable.ost.api.model.impl.Location.StationType types[]) throws IOException;
}

File: lib/common/src/main/java/ch/unstable/ost/api/StationsDAO.kt
package ch.unstable.ost.api


import java.io.IOException

import ch.unstable.ost.api.model.Station

interface StationsDAO {
    @Throws(IOException::class)
    fun getStationsByQuery(query: String, types: List<Station.StationType> = listOf()): List<Station>
}
